docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+1. <b>POST /adapter</b> - to get a job result from a CL node
+2. <b>POST /jobs</b> - to get a new job addition from a CL node
+3. <b>DELETE /jobs/:id</b> - to get a job removal from a CL node
+
+What it does:
+
+<div id='postadapter'></div>
+
+a. <u>/adapter</u>: This is the endpoint which is used to accept job results from CL job requests. This endpoint does the following:
+  - Read the result, request ID, request Type and job name from the received body
+  - Check if an actual result was received and if not return a status 500 response. If a result is received, a 200 is sent as response.
+  - Obtain the last price on-chain from the state
+  - Obtain the round to push the price for by checking whether the latest round on-chain is greater than the oracle's last reported round. If so, the round ID would be the latest round on chain. Otherwise, the latest round on chain is incremented by 1 as a new round would need to be created.
+  - Push the price on chain if it satisfies all the following criteria:
+    - If it is a new round and we the oracle has not started the current on-chain round or if it is not a new round but the oracle has not submitted to the current round on-chain
+    - One of the following:
+      - If its time for a price update by comparing the timestamp of the last round and now by making use of <b>pushInterval</b>(for that feed) (Request type 1)
+      - If there is a price deviation greater than <b>priceDeviationPerc</b>(for that feed) between the received price and the latets price on chain (Request type 2)
+      - If there was a new a new round (Request type 3)
+  - Update the 'last_reported_round' and the request ID of latest job result received from CL node in the state file
+<div id='postjobs'></div>
+
+b. <u>/jobs</u>: This is the endpoint which is used to handle a new job created on the CL node. This endpoint adds the job to the list of jobs in the state and intialises all the required states.
+
+<div id='deljobs'></div>
+
+c. <u>/jobs/:id</u>:  This is the endpoint which is used to handle a job removal on the CL node. This endpoint removes the job from the list of jobs in the state and removes any related states.
+
+<br>
+<div id='middlewarefunc'></div>
+
+<b>middleware()</b>
+
+Use: This function is used as entry to the middleware functions
+
+What it does:
+  2. Initialises state
+  3. Starts the bridge to listen for new jobs and job results
+  4. Starts the controller on the first second of the next minute. This is done to try and have all middleware pushing jobs on approximetely the first second of each new minute
+
+<br>
+<div id='monitorjs'></div>
+
+### <u>monitor.js</u>
+
+This file contains all the functions which are used both by the monitoring script. 
+
+<br>
+<div id='envvarsmonitor'></div>
+
+<b>Environment Variables</b>
+
+This script makes use of the following environment variables and it requires them in order to function. In fact, it contains validation upon entry to make sure they are well defined.
```

Comment:
You're saying that the env variables are required but then assign a default value to them?

Reply:
Removed the required part in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/66a78fb742ad2bef38d1b3a84d13bfa1c0811e15